includes/class-wc-payments-account.php

```diff
@@ -584,11 +600,21 @@ public function maybe_redirect_to_server_link() {
 		$args = $_GET;
 		unset( $args['wcpay-link-handler'] );
 
+		$this->redirect_to_account_link();
+	}
+
+	/**
+	 * Function to immediately redirect to the account link.
+	 *
+	 * @param bool $is_complete_kyc_link Optional indicates if is a link to continue kyc process.
+	 */
+	private function redirect_to_account_link( bool $is_complete_kyc_link = false ) {
 		try {
-			$link = $this->payments_api_client->get_link( $args );
-			if ( isset( $args['type'] ) && 'complete_kyc_link' === $args['type'] && isset( $link['state'] ) ) {
-				set_transient( 'wcpay_stripe_onboarding_state', $link['state'], DAY_IN_SECONDS );
```

Comment:
I did a quick check and it seems like this transient gets checked in a few places ([ref](https://github.com/Automattic/woocommerce-payments/blob/update/disable-apms-po/includes/class-wc-payments-account.php#L1247)) - would we still need to set it in the case of a `complete_kyc_link` to make sure this logic works properly?

Reply:
Thanks Dan! I didn't intend to remove that set transient, thanks for catching it up :D Readded here https://github.com/Automattic/woocommerce-payments/pull/7009/commits/171bc4aa3f79a1c9cf49170f70332d0991f7d629